src/wallet/spend.cpp

```diff
@@ -897,7 +897,9 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
     // allowed (coins automatically selected by the wallet)
     CoinsResult available_coins;
     if (coin_control.m_allow_other_inputs) {
-        available_coins = AvailableCoins(wallet, &coin_control, coin_selection_params.m_effective_feerate);
+        CoinFilterParams params; // include non-spendable coins if this is a watch-only wallet
+        params.only_spendable = !wallet.IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS);
```

Comment:
Can we test for this?

Reply:
Actually, this change is not needed here. Will just drop it.
I was using it to add support for creating PSBTs on legacy wallets that only know the pubkey hash and not the pubkey (in other terms, use `importaddress` instead of `importpubkey`). Which has never being possible.